source/video_core/src/core/hw/gpu.cpp

```diff
@@ -226,6 +226,12 @@ inline void Write(Pica::Context& context, u32 addr, const T data) {
                                                             config.display_transfer.input_height.Value(),
                                                             config.display_transfer.output_height.Value()));
                 }
+                
+                if (config.flip_data && (config.display_transfer.output_height != config.display_transfer.input_height)) {
+                    throw std::runtime_error(fmt::format(   "Flipping Y; Input height {:#x} doesn't match output height {:#x} exactly",
```

Comment:
The code you copied from predates `Mikage::Exceptions`, but nowadays NotImplemented should be thrown to highlight it's a possibly valid but untested+unimplemented configuration:
```suggestion
                    throw Mikage::Exceptions::NotImplemented(   "Flipping Y; Input height {:#x} doesn't match output height {:#x} exactly",
```

(Just for further context: This is in contrast to the `>` check above, which would use `Invalid`, since it's a hardware configuration that's assumed to not be used by games)

Reply:
That's some good communication, I like it.